src/toil/fileStore.py

```diff
@@ -851,7 +851,8 @@ def cleanCache(self, newJobReqs):
                 #                  (self.decodedFileID(cachedFile), cachedFileSize))
                 logger.debug('CACHE: Evicted  file with ID \'%s\' (%s bytes)' %
                              (self.decodedFileID(cachedFile), cachedFileSize))
-            assert cacheInfo.isBalanced(), 'Unable to free up enough space for caching.'
```

Comment:
That's a very important error. We could make it more verbose but really, we should never see this issue.